src/appleseed/foundation/utility/unzipper.cpp

```diff
@@ -0,0 +1,217 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2010-2013 Francois Beaune, Jupiter Jazz Limited
+// Copyright (c) 2014-2017 Francois Beaune, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+
+// Interface header.
+#include "unzipper.h"
+
+// appleseed.foundation headers.
+#include "foundation/core/exceptions/exception.h"
+#include "foundation/utility/minizip/unzip.h"
+
+// Boost headers.
+#include "boost/filesystem.hpp"
+
+// Standard headers.
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <sstream>
+
+using namespace std;
+namespace bf = boost::filesystem;
+
+namespace foundation
+{
+
+const int BUFFER_SIZE = 4096;
+
+const char* UnzipException::what() const throw() 
+{
+    stringstream exc_message;
+    exc_message << message;
+
+    if (err != 0)
+        exc_message << err;
+
+    return exc_message.str().c_str();
+}
+
+bool exists(const string& dirname) {
```

Comment:
I don't think these helper methods are really useful. I would recommend using Boost.Filesystem directly as this will expose the actual Boost methods you're using, and hence will make the exact semantics of your code clearer to readers.